Storyblok Revision: Home page and blogPost filtering (#2579)
withastro/docs: src/pages/en/guides/cms/storyblok.mdx

### Fetching data

To test the setup, create a new page in the `src/pages` directory named `index.astro` with the following content:

> ```astro title="pages/index.astro"

Reviewer: ``````suggestion
```astro title="src/pages/index.astro"
``````
Pretty sure we mostly start filepaths from `src/`.

---
import { useStoryblokApi } from '@storyblok/astro'
import StoryblokComponent from '@storyblok/astro/StoryblokComponent.astro'

const storyblokApi = useStoryblokApi()

const { data } = await storyblokApi.get("cdn/stories/home", {
  version: import.meta.env.DEV ? "draft" : "published",
});

> 

Reviewer 2: Maybe we need a disclaimer at the end stating that this fetch will only work if you have a story named `home`

Author: We do say to create `home` earlier

Author: My bad - I meant to only make this change in the **Making a blog** section. Reverting this one

### Generating pages

> To create a route for a specific `page`, you can fetch its directly from the Storyblok API and pass it to the `Page` component. Create an `index.astro` file in `src/pages` to render the `home` page:

Reviewer: ```suggestion
To create a route for a specific `page`, you can fetch its content directly from the Storyblok API and pass it to the `Page` component. Create an `index.astro` file in `src/pages/` to render the `home` page:
```
Fetch "it", or fetch "its....(missing word/property)"? (I'd lean towards the latter, because you have to pass an actual *thing* to the `<Page />` component.

(This suggestion only edits what's there, and does not take into account Kevin's comment that perhaps the step should itself be a different instruction. 😄 )

```astro title="src/pages/index.astro" {7,8,15} 
---
import { useStoryblokApi } from '@storyblok/astro'
import BaseLayout from '../layouts/BaseLayout.astro'
import Page from '../storyblok/Page.astro';

const storyblokApi = useStoryblokApi();
const { data } = await storyblokApi.get('cdn/stories/home');
const content = data.story.content;
---
<html lang="en">
  <head>
    <title>Storyblok & Astro</title>
  </head>
  <body>

>     <Page blok={content} />

Reviewer 2: ```suggestion
    <Page blok={content} />
```

I am not sure why we are not using the Storyblok component (from the storyblok library) in this case. It should work as it was but only fetching data from one story.

Author: What is `Page` for then? I thought `Page` was for rendering the `page` type

Reviewer 2: @Jutanium I see where this is coming from. Uhmm I am not sure which one should be used. Page is the container for multiple bloks in the page content-type story.

  </body>
</html>
```

> To dynamically render all of the blog posts, create a dynamic route. This approach varies depending on whether you're using **static site generation** (the default) or **server-side rendering**.

Reviewer: ```suggestion
To generate pages for all of your blog posts, create a `.astro` page that will create dynamic routes. This approach varies depending on whether you're using **static site generation** (the default) or **server-side rendering**.
```
Just thinking that the reader doesn't care that "blog posts are dynamically generated" so much as they care that "a page is made for each blog post"? Plus, this avoids using the word "dynamic" twice. :smile: (And, I'm not sure I'd say *a* dynamic route is created, but a page that creates a bunch of routes.) Tweak this as you like!

#### Static site generation

If you are using Astro's default static site generation, you will use [dynamic routes](/en/core-concepts/routing/#dynamic-routes) and the `getStaticPaths` function to generate your project pages.

> Create a new file in a `src/pages/blog` directory called `[...slug].astro` and add the following code:

Reviewer: ```suggestion
Create a new directory `src/pages/blog/` and add a new file called `[...slug].astro` with the following code:
```
If the directory doesn't exist, probably good practice to instruct the reader to create the directory.

> If you’ve [opted into SSR mode](/en/guides/server-side-rendering/#enabling-ssr-in-your-project), you will use dynamic routes to fetch the page data from Storyblok.

Reviewer: Just noting here that the previous instruction to create the directory and the file is techically under "if you're using SSG" so you might want a similar instruction (vs just a code sample that exists in comparison to the previous section) in case someone just skimmed and came directly here.
